third_party/gpus/cuda_configure.bzl

```diff
@@ -120,13 +120,33 @@ def _enable_cuda(repository_ctx):
   return False
 
 
-def _cuda_toolkit_path(repository_ctx):
-  """Finds the cuda toolkit directory."""
+def _cuda_toolkit_path(repository_ctx, cuda_version):
+  """Finds the cuda toolkit directory.
+
+   If cuda_version is given, attempts to return a real path.
+
+  Args:
+    repository_ctx: The repository context.
+    cuda_version: The cuda toolkit version.
+
+  Returns:
+    A speculative real path of the cuda toolkit install directory.
+  """
   cuda_toolkit_path = _DEFAULT_CUDA_TOOLKIT_PATH
   if _CUDA_TOOLKIT_PATH in repository_ctx.os.environ:
     cuda_toolkit_path = repository_ctx.os.environ[_CUDA_TOOLKIT_PATH].strip()
   if not repository_ctx.path(cuda_toolkit_path).exists:
     auto_configure_fail("Cannot find cuda toolkit path.")
+
+  if cuda_version:
+    # Handle typical configuration where the real path is
+    # <basedir>/cuda-<version> and the provided path is <basedir>/cuda.
+    version_suffixed = "%s-%s" % (cuda_toolkit_path, cuda_version)
+    if repository_ctx.path(version_suffixed).exists:
+      return version_suffixed
+  # Now hoping that the provided path was already a real path.
```

Comment:
Returns the non-versioned path if cuda version is not provided or if the installation does not use a cuda-<version> directory, such as on ArchLinux where CUDA installs directly to /opt/cuda.